src/services/SlackEventsService.ts

```diff
@@ -0,0 +1,74 @@
+import { generateSlackSignature, slack } from "../utils";
+import Express from "express";
+
+interface Event {
+  ts: string;
+  text: string;
+  channel: string;
+  event_ts: string;
+  thread_ts?: string;
+}
+
+class SlackEventsService {
+  async appMention(req: Express.Request, res: Express.Response) {
+    const event = req.body.event as Event;
+    const { channel, ts } = event;
+
+    try {
+      const message = await slack.chat.postMessage({
+        channel,
+        thread_ts: ts,
+        text: "I'm thinking...",
+      });
+
+      const sendMessageEndpointUrl = `${req.protocol}://${req.hostname}/send-message`;
+
+      // send message in background, this is for slack's 3 second response time limit
+      const sendMessageBody = JSON.stringify({
+        event: {
+          channel,
+          thread_ts: message.message.thread_ts,
+          ts: message.message.ts,
+          text: event.text,
+        },
+      });
+
+      const slackRequestTimestamp = req.headers[
+        "x-slack-request-timestamp"
+      ] as string;
+
+      const slackSignature = generateSlackSignature(
+        slackRequestTimestamp,
+        sendMessageBody
+      );
+      // send acknowledgement request to slack that we recieved the message
+
+      try {
+        fetch(sendMessageEndpointUrl, {
+          method: "post",
+          headers: {
+            "Content-Type": "application/json",
+            "x-slack-request-timestamp": slackRequestTimestamp,
+            "x-slack-signature": slackSignature,
+          },
+          body: sendMessageBody,
+        });
+        console.log("Request sent successfully!");
+      } catch (error) {
+        console.error("Error sending request:", error);
+      }
+      // a hacky way to ensure the request is sent
+      await new Promise((resolve) => setTimeout(resolve, 10));
```

Comment:
@homanp  this is not ideal, but seems to be easiest way for now.

The problem is since getting answer from Superagent API is relatively long running process and Slack bot needs to send response back to Slack API within the 3 seconds, otherwise it will timeout. So I created another endpoint to handle the getting answer from Superagent and sending request to that endpoint